tests/unit/test_network_download.py

```diff
@@ -350,3 +351,67 @@ def test_downloader(
 
     # Make sure that the downloader makes additional requests for resumption
     _http_get_mock.assert_has_calls(calls)
+
+
+def test_resumed_download_caching(tmpdir: Path) -> None:
+    """Test that resumed downloads are cached properly for future use."""
+    session = PipSession()
+    link = Link("http://example.com/foo.tgz")
+    downloader = Downloader(session, "on", resume_retries=5)
+
+    # Mock an incomplete download followed by a successful resume
+    incomplete_resp = MockResponse(b"0cfa7e9d-1868-4dd7-9fb3-")
+    incomplete_resp.headers = {"content-length": "36"}
+    incomplete_resp.status_code = 200
+
+    resume_resp = MockResponse(b"f2561d5dfd89")
+    resume_resp.headers = {"content-length": "12"}
+    resume_resp.status_code = 206
+
+    responses = [incomplete_resp, resume_resp]
+    _http_get_mock = MagicMock(side_effect=responses)
+
+    # Mock the session's adapters to have a cache controller
+    mock_adapter = MagicMock(spec=CacheControlAdapter)
+    mock_controller = MagicMock()
+    mock_adapter.controller = mock_controller
+    mock_controller.cache_url = MagicMock(return_value="cache_key")
+    mock_controller.serializer = MagicMock()
+    mock_controller.serializer.dumps = MagicMock(return_value=b"serialized_data")
+
+    # Mock the cache to be a SafeFileCache
+    mock_cache = MagicMock(spec=SafeFileCache)
+    mock_adapter.cache = mock_cache
```

Comment:
Could you just not create a pip session with caching already enabled? `PipSession(cache=str(tmpdir))` I get this is an unit test, but practically, I'd prefer if the we tested the actual implementation to avoid mismatches that hide issues down the line.

Reply:
Yeah, I'm now using a `PipSession` object to test here.